### PR TITLE
Make app buildable on MacOS and portable

### PR DESCRIPTION
**Pull request by ArmandMeppa**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -3,7 +3,9 @@ name: Build Binaries
 on:
   push:
     branches:
-      - main
+      - dev
```
Comment (stephane-segning) [inline .github/workflows/build.yaml:6]: Why dev?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -16,8 +18,15 @@ env:
   MAINTAINER_EMAIL: Armand.Meppa@adorsys.com
```
Comment (stephane-segning) [inline .github/workflows/build.yaml:18]: Why the need of the maintenaner email?

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -39,49 +48,48 @@ jobs:
...
+          name: ${{ env.APP_NAME }}-${{ matrix.target }}
+          path: ./
+
+  release:
+    name: Release
+    permissions: write-all
+    needs:
+      - build-ubuntu
+      # - build-macos
+      # - build-windows
```
Comment (stephane-segning) [inline .github/workflows/build.yaml:65]: Remove the comments if not used

File: main.go [hunk 4]
```diff
@@ -12,27 +12,34 @@ import (
...
-	case "linux":
-		iconPath = "./assets/wazuh-logo-min.png"
+	case "linux", "darwin":
+		iconPath = "assets/wazuh-logo-min.png"
 	case "windows":
-		iconPath = "./assets/wazuh-logo-min.ico"
+		iconPath = "assets/wazuh-logo-min.ico"
 	default:
-		iconPath = "./assets/wazuh-logo-min.png" // Fallback to PNG for other OS
+		iconPath = "assets/wazuh-logo-min.png" // Fallback to PNG for other OS
```
Comment (stephane-segning) [inline main.go:33]: This means that windows is the only special case.

File: scripts/install.sh [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    set -euo pipefail
+else
+    set -eu
+fi
+
+# Default log level and application details
+LOG_LEVEL=${LOG_LEVEL:-INFO}
+APP_NAME=${APP_NAME:-"wazuh-agent-status"}
+WOPS_VERSION=${WOPS_VERSION:-"0.2.1"}
+OSSEC_CONF_PATH=${OSSEC_CONF_PATH:-"/var/ossec/etc/ossec.conf"}
```
Comment (stephane-segning) [inline scripts/install.sh:14]: Why the need of this variable ?

File: scripts/install.sh [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+if [ -n "$BASH_VERSION" ]; then
+    set -euo pipefail
+else
+    set -eu
+fi
+
+# Default log level and application details
+LOG_LEVEL=${LOG_LEVEL:-INFO}
+APP_NAME=${APP_NAME:-"wazuh-agent-status"}
+WOPS_VERSION=${WOPS_VERSION:-"0.2.1"}
```
Comment (stephane-segning) [inline scripts/install.sh:13]: Why the need of this variable ?

**Review by stephane-segning — CHANGES_REQUESTED** [review]
No build for macos and windows?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ArmandMeppa) [reply .github/workflows/build.yaml:6]: It was for testing purposes, but ok

File: .github/workflows/build.yaml (hunk 2 above)
Comment (ArmandMeppa) [reply .github/workflows/build.yaml:18]: It was to create a *.deb* file. But with the new way, it's no more useful

File: .github/workflows/build.yaml (hunk 3 above)
Comment (ArmandMeppa) [reply .github/workflows/build.yaml:65]: They will be used later, after the linux version will be stabilised

File: main.go (hunk 4 above)
Comment (ArmandMeppa) [reply main.go:33]: One never knows on which system a user can run this script, but I get the point

File: scripts/install.sh (hunk 6 above)
Comment (ArmandMeppa) [reply scripts/install.sh:13]: This file isn't the point of interest for now.

**Comment by ArmandMeppa** [comment]
> No build for macos and windows?

It's not the definitive version, I want us to test if the app works on those platform using 
```sh
go run .
```
or 
```sh
go build -o dist/
./dist/wazuh-agent-status
```

**Comment by bengo237** [comment]
I run this :
``` CGO_ENABLED=1 GOOS=windows GOARCH=amd64 go build -o dist/wazuh-tray.exe```
after generate the wazuh-tray.exe copy on windows os and try to run but it doesn't work but the linux one work perfectly

File: .github/workflows/build.yaml (hunk 3 above)
Comment (stephane-segning) [reply .github/workflows/build.yaml]: Meaning that you could not test locally using arc?

Comment (ArmandMeppa) [reply .github/workflows/build.yaml]: Don't have it yet locally

File: main.go [hunk 7]
```diff
@@ -12,27 +12,32 @@ import (
...
-	case "linux":
-		iconPath = "./assets/wazuh-logo-min.png"
+	case "linux", "darwin":
+		iconPath = "assets/wazuh-logo-min.png"
 	case "windows":
-		iconPath = "./assets/wazuh-logo-min.ico"
-	default:
-		iconPath = "./assets/wazuh-logo-min.png" // Fallback to PNG for other OS
+		iconPath = "assets/wazuh-logo-min.ico"
+	}
```
Comment (stephane-segning) [inline main.go:32]: Just use default for the other OS, and handle `windows` as a special case

**Comment by ArmandMeppa** [comment]
> I run this : ` CGO_ENABLED=1 GOOS=windows GOARCH=amd64 go build -o dist/wazuh-tray.exe` after generate the wazuh-tray.exe copy on windows os and try to run but it doesn't work but the linux one work perfectly

Please run them on MacOS and Windows, using this command 
```sh
go run .
```

File: scripts/install.sh [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+curl -SL --progress-bar -o "$TEMP_DIR/$BIN_NAME" "$URL" || error_exit "Failed to download $BIN_NAME"
+
+# Step 2: Install the binary
+print_step 2 "Installing binary to $BIN_DIR..."
+maybe_sudo mkdir -p "$BIN_DIR" || error_exit "Failed to create directory $BIN_DIR"
+maybe_sudo mv "$TEMP_DIR/$BIN_NAME" "$BIN_DIR/$APP_NAME" || error_exit "Failed to move binary to $BIN_DIR"
+maybe_sudo chmod 750 "$BIN_DIR/$APP_NAME" || error_exit "Failed to set executable permissions on the binary"
+
+success_message "Installation and configuration complete! You can now use '$BIN_DIR/$APP_NAME' from your terminal."
+info_message "Run \n\n\t${GREEN}${BOLD}sudo $BIN_DIR/$APP_NAME o-auth2${NORMAL}\n\n to start configuring. If you don't have sudo on your machine, you can run the command without sudo."
```
Comment (stephane-segning) [inline scripts/install.sh:112]: Here I can see the steps are 
1. Download the bin
2. Install the bin.

How's the user running it? Is he running it manually? Or is it starting by itself?

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Comment by ArmandMeppa** [comment]
I am looking for ways to make macOS trust my binary file, programmatically

**Pull request merged**